tools/linter/adapters/set_linter/configs.py

```diff
@@ -0,0 +1,41 @@
+import argparse
+import json
+
+
+def add_configs(args: argparse.Namespace, filename: str) -> argparse.Namespace:
+    try:
+        with open(filename) as fp:
+            config = json.load(fp)
+    except FileNotFoundError:
+        config = {}
+
+    if bad := set(config) - set(vars(args)):
+        s = "" if len(bad) == 1 else "s"
+        bad_name = ", ".join(sorted(bad))
+        raise ValueError(f"Unknown arg{s}: {bad_name}")
+
+    for k, v in config.items():
+        if k == "fix" and args.fix is None:
+            args.fix = v
+        else:
+            setattr(args, k, getattr(args, k) or v)
+
+    return args
+
+
+def make_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
```

Comment:
Suggestion: Add an extended help explaining some rough edges like the issues we discussed with `fix` mode and typing!